pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableImplV3.java

```diff
@@ -175,8 +175,11 @@ public DataTableImplV3(ByteBuffer byteBuffer)
   }
 
   @Override
-  public void addException(ProcessingException processingException) {
-    _errCodeToExceptionMap.put(processingException.getErrorCode(), processingException.getMessage());
+  public void addException(Exception exception) {
+    if (exception instanceof ProcessingException) {
+      ProcessingException processingException = (ProcessingException) exception;
+      _errCodeToExceptionMap.put(processingException.getErrorCode(), processingException.getMessage());
```

Comment:
This is a workaround for the fact that. ProcessingException is a generated class that we can't include in spi --> because it will pull in unnecessary dependencies to the SPI module

So instead I opted to relax the interface signature but tighten it during impl.